gui/default/syncthing/core/syncthingController.js

```diff
@@ -1546,6 +1551,13 @@ angular.module('syncthing.core')
             return 'text';
         };
 
+        $scope.changedFolderLabel = function(){
```

Comment:
Ah, this I don't like. :) It results in folder IDs like `7DDRT7J-UICR4PM-ILLYJLBV` (which I think is a tad too long; the first two words would do) where only the last part changes. So all folder IDs created by anyone is going to start with `7DDRT7J-UICR4PM` for a long time, making them look the same to humans. Also there's no need to change the folder ID for every character typed. I suggest simply generating a fully random folder ID when the create-folder dialog is opened and then let it be. It could be set when creating the $scope.currentFolder.